client/src/modules/list/model/actions.js

```diff
@@ -373,14 +374,24 @@ export const addListViewer = (listId, email) => dispatch =>
   patchData(`/api/lists/${listId}/add-viewer`, {
     email
   })
-    .then(resp => resp.json())
+    .then(resp => {
+      if (resp.status === 200) {
+        return resp.json();
+      }
+    })
     .then(json => {
-      dispatch(addViewerSuccess(json, listId));
-      createNotificationWithTimeout(
-        dispatch,
-        NotificationType.SUCCESS,
-        json.message || 'Viewer added successfully.'
-      );
+      if (json) {
+        dispatch(addViewerSuccess(json, listId));
+        createNotificationWithTimeout(
+          dispatch,
+          NotificationType.SUCCESS,
+          json.message || 'Viewer added successfully.'
+        );
+
+        return UserCreationStatus.CREATED;
```

Comment:
You don't create the user here - You add/invite the user to the list:
`UserAddingStatus.ADDED` or `UserInvitingStatus.INVITED`